client/components/date-range/test/index.js

```diff
@@ -297,225 +253,202 @@ describe( 'DateRange', () => {
+			// Dates on/before `10-03-2018`
+			someEnabledDatesStrings.forEach( ( dateString ) => {
+				const el = screen.getByLabelText( dateString );
+				expect( el ).toHaveAttribute( 'aria-disabled', 'false' );
+			} );
 
-			expect( dateRangeInputs.props() ).toEqual(
-				expect.objectContaining( {
-					startDateValue: dateToLocaleString( newStartDate ),
-					endDateValue: dateToLocaleString( newEndDate ),
-				} )
-			);
+			// Dates after `10-03-2018`
+			someDisabledDatesStrings.forEach( ( dateString ) => {
+				const el = screen.getByLabelText( dateString );
+				expect( el ).toHaveAttribute( 'aria-disabled', 'true' );
+			} );
 		} );
 
-		test( 'should update start date selection on start date input blur event', () => {
-			const endDate = moment( '2018-05-30' );
-			const startDate = moment( endDate ).subtract( 1, 'months' );
+		test( 'should *not* show the navigation button for the month prior to the one for `firstSelectableDate`', async () => {
+			const firstSelectableDate = moment( '10-03-2018', 'MM-DD-YYYY' );
 
-			const wrapper = shallow(
+			render(
 				<DateRange
-					selectedStartDate={ startDate }
-					selectedEndDate={ endDate }
 					translate={ translate }
 					moment={ moment }
+					firstSelectableDate={ firstSelectableDate }
 				/>
 			);
 
-			wrapper.instance().handleInputBlur( '03/20/2018', 'Start' );
+			await userEvent.click( screen.getByLabelText( 'Select date range' ) );
 
-			expect( dateToLocaleString( wrapper.state().startDate ) ).toEqual(
-				dateToLocaleString( '03/20/2018' )
-			);
+			const previousMonthBtnEl = screen.queryByLabelText( /Previous month/ );
+			const nextMonthBtnEl = screen.getByLabelText( 'Next month (December 2018)' );
+
+			expect( previousMonthBtnEl ).toBeNull();
+			expect( nextMonthBtnEl ).toBeVisible();
 		} );
 
-		test( 'should update end date selection on end date input blur event', () => {
-			const endDate = moment( '2018-05-30' );
-			const startDate = moment( endDate ).subtract( 1, 'months' );
+		test( 'should *not* show the navigation button for the month afer to the one for `lastSelectableDate`', async () => {
+			const lastSelectableDate = moment( '10-03-2018', 'MM-DD-YYYY' );
 
-			const wrapper = shallow(
+			render(
 				<DateRange
-					selectedStartDate={ startDate }
-					selectedEndDate={ endDate }
 					translate={ translate }
 					moment={ moment }
+					lastSelectableDate={ lastSelectableDate }
 				/>
 			);
 
-			const inputs = wrapper.find( DateRangeInputs );
+			await userEvent.click( screen.getByLabelText( 'Select date range' ) );
 
-			inputs.props().onInputBlur( '05/28/2018', 'End' );
+			const previousMonthBtnEl = screen.getByLabelText( 'Previous month (August 2018)' );
+			const nextMonthBtnEl = screen.queryByLabelText( /Next month/ );
 
-			expect( dateToLocaleString( wrapper.state().endDate ) ).toEqual(
-				dateToLocaleString( '05/28/2018' )
-			);
+			expect( previousMonthBtnEl ).toBeVisible();
+			expect( nextMonthBtnEl ).toBeNull();
```

Comment:
Same as above.

Reply:
Done in https://github.com/Automattic/wp-calypso/pull/63879/commits/b50bb71c4c0bb51094e7b6e4ef076ba3ac5489d1.